Makefile

```diff
@@ -607,8 +627,44 @@ $(BUILD_DIR)/dmadata/dmadata_table_spec.h $(BUILD_DIR)/dmadata/compress_ranges.t
 $(BUILD_DIR)/src/boot/z_std_dma.o: $(BUILD_DIR)/dmadata/dmadata_table_spec.h
 $(BUILD_DIR)/src/dmadata/dmadata.o: $(BUILD_DIR)/dmadata/dmadata_table_spec.h
 
-$(BUILD_DIR)/%.o: %.s
-	$(CPP) $(CPPFLAGS) $(IINC) $< | $(AS) $(ASFLAGS) $(IINC) $(ENDIAN) -o $@
+$(BUILD_DIR)/asm/%.o: asm/%.s
+	$(AS) $(ASFLAGS) $(IINC) $(ENDIAN) $< -o $@
+
+$(BUILD_DIR)/data/%.o: data/%.s
+	$(AS) $(ASFLAGS) $(IINC) $(ENDIAN) $< -o $@
+
+$(BUILD_DIR)/rsp/%.o: rsp/%.s
+	$(AS) $(ASFLAGS) $(IINC) $(ENDIAN) $< -o $@
+
+# Assemble the ROM header with GNU AS always
+$(BUILD_DIR)/src/makerom/rom_header.o: src/makerom/rom_header.s
+ifeq ($(COMPILER),ido)
+	$(CPP) $(CPPFLAGS) $(MIPS_BUILTIN_DEFS) $(IINC) $< | $(AS) $(ASFLAGS) $(IINC) $(ENDIAN) -o $@
+else
+	$(CCAS) -c $(CCASFLAGS) $(IINC) $(MIPS_VERSION) $(ASOPTFLAGS) -o $@ $<
+endif
+	$(OBJDUMP_CMD)
+
+$(BUILD_DIR)/src/makerom/ipl3.o: $(EXTRACTED_DIR)/incbin/ipl3
+	$(OBJCOPY) -I binary -O elf32-big --rename-section .data=.text $< $@
```

Comment:
What's the advantage of using objcopy over incbin assembly files?

Reply:
incbin isn't supported by IDO, so we'd need custom makefile rule anyway. At that point, objcopy is simpler